lang/c++/impl/NodeImpl.cc

```diff
@@ -318,11 +319,33 @@ void NodePrimitive::printDefaultToJson(const GenericDatum &g, std::ostream &os,
       os << g.value<int64_t>();
       break;
     case AVRO_FLOAT:
-      os << g.value<float>();
-      break;
+      {
+          auto val = g.value<float>();
+          if (boost::math::isnan(val)) {
+              os << "\"NaN\"";
```

Comment:
does this class have test cases? if so, please can tests be added.

Reply:
Added a test case and existing test cases also fail without this change (since test schemas are using NaN, Infinity default values)